### ISSUE TITLE
How to Implement MiniDrawer

### ISSUE THREAD

**Issue by sendtodilanka** [mikepenz/MaterialDrawer#1429]
I have tried to implement MiniDrawer. It's give result as below picture. how can i solve?

![screenshot_2016-07-30-20-08-43](https://cloud.githubusercontent.com/assets/20735375/17271175/b2d34ed2-5692-11e6-9b1d-a0edecd13c28.png)

This is MainActivity.Java

```java
    public class MainActivity extends AppCompatActivity {

    private Drawer result = null;
    private MiniDrawer miniResult = null;
    private Crossfader crossFader;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });

        getSupportActionBar().setTitle(R.string.drawer_item_mini_drawer);

        result = new DrawerBuilder()
                .withActivity(this)
                .withToolbar(toolbar)
                .withTranslucentNavigationBar(false)
                .addDrawerItems(
                        new PrimaryDrawerItem().withName(R.string.drawer_item_compact_header).withIcon(GoogleMaterial.Icon.gmd_wb_sunny).withIdentifier(1),
                        new PrimaryDrawerItem().withName(R.string.drawer_item_action_bar_drawer).withIcon(FontAwesome.Icon.faw_home).withBadge("22").withBadgeStyle(new BadgeStyle(Color.RED, Color.RED)).withIdentifier(2).withSelectable(false),
                        new PrimaryDrawerItem().withName(R.string.drawer_item_multi_drawer).withIcon(FontAwesome.Icon.faw_gamepad).withIdentifier(3),
                        new PrimaryDrawerItem().withName(R.string.drawer_item_non_translucent_status_drawer).withIcon(FontAwesome.Icon.faw_eye).withIdentifier(4),
                        new PrimaryDrawerItem().withDescription("A more complex sample").withName(R.string.drawer_item_advanced_drawer).withIcon(GoogleMaterial.Icon.gmd_adb).withIdentifier(5),
                        new SectionDrawerItem().withName(R.string.drawer_item_section_header),
                        new SecondaryDrawerItem().withName(R.string.drawer_item_open_source).withIcon(FontAwesome.Icon.faw_github),
                        new SecondaryDrawerItem().withName(R.string.drawer_item_contact).withIcon(GoogleMaterial.Icon.gmd_format_color_fill).withTag("Bullhorn"),
                        new DividerDrawerItem(),
                        new SwitchDrawerItem().withName("Switch").withIcon(Octicons.Icon.oct_tools).withChecked(true).withOnCheckedChangeListener(onCheckedChangeListener),
                        new ToggleDrawerItem().withName("Toggle").withIcon(Octicons.Icon.oct_tools).withChecked(true).withOnCheckedChangeListener(onCheckedChangeListener)
                ) // add the items we want to use with our Drawer
                .withOnDrawerItemClickListener(new Drawer.OnDrawerItemClickListener() {
                    @Override
                    public boolean onItemClick(View view, int position, IDrawerItem drawerItem) {
                        if (drawerItem instanceof Nameable) {
                            Toast.makeText(MainActivity.this, ((Nameable) drawerItem).getName().getText(MainActivity.this), Toast.LENGTH_SHORT).show();
                        }
                        return false;
                    }
                })
                .withGenerateMiniDrawer(true)
                .withSavedInstance(savedInstanceState)
                .buildView();

        miniResult = result.getMiniDrawer();

        int firstWidth = (int) UIUtils.convertDpToPixel(300, this);
        int secondWidth = (int) UIUtils.convertDpToPixel(72, this);

        crossFader = new Crossfader()
                .withContent(findViewById(R.id.main_content))
                .withFirst(result.getSlider(), firstWidth)
                .withSecond(miniResult.build(this), secondWidth)
                .withSavedInstance(savedInstanceState)
                .build();

        miniResult.withCrossFader(new CrossfadeWrapper(crossFader));

        //define a shadow (this is only for normal LTR layouts if you have a RTL app you need to define the other one
        crossFader.getCrossFadeSlidingPaneLayout().setShadowResourceLeft(R.drawable.material_drawer_shadow_left);
    }

    private OnCheckedChangeListener onCheckedChangeListener = new OnCheckedChangeListener() {
        @Override
        public void onCheckedChanged(IDrawerItem drawerItem, CompoundButton buttonView, boolean isChecked) {
            if (drawerItem instanceof Nameable) {
                Log.i("material-drawer", "DrawerItem: " + ((Nameable) drawerItem).getName() + " - toggleChecked: " + isChecked);
            } else {
                Log.i("material-drawer", "toggleChecked: " + isChecked);
            }
        }
    };

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        //add the values which need to be saved from the drawer to the bundle
        outState = result.saveInstanceState(outState);
        //add the values which need to be saved from the crossFader to the bundle
        outState = crossFader.saveInstanceState(outState);
        super.onSaveInstanceState(outState);
    }

    @Override
    public void onBackPressed() {
        //handle the back press :D close the drawer first and if the drawer is closed close the activity
        if (crossFader != null && crossFader.isCrossFaded()) {
            crossFader.crossFade();
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        menu.findItem(R.id.menu_1).setIcon(new IconicsDrawable(this, FontAwesome.Icon.faw_repeat).color(Color.WHITE).actionBar());
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        switch (item.getItemId()){
            case R.id.menu_1:
                crossFader.crossFade();
                return true;
            case R.id.act_settings:
                return true;
            default:
        }

        return super.onOptionsItemSelected(item);
    }
    }
``` 
This is activity_main.xml

```xml
<?xml version="1.0" encoding="utf-8"?>
          <android.support.design.widget.CoordinatorLayout                  xmlns:android="http://schemas.android.com/apk/res/android"
         xmlns:app="http://schemas.android.com/apk/res-auto"
         xmlns:tools="http://schemas.android.com/tools"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:fitsSystemWindows="true"
         tools:context="com.sldroids.minidrawer_v2.MainActivity">

        <android.support.design.widget.AppBarLayout
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:theme="@style/AppTheme.AppBarOverlay">

        <android.support.v7.widget.Toolbar
            android:id="@+id/toolbar"
            android:layout_width="match_parent"
            android:layout_height="?attr/actionBarSize"
            android:background="?attr/colorPrimary"
            app:popupTheme="@style/AppTheme.PopupOverlay" />

    </android.support.design.widget.AppBarLayout>

    <include layout="@layout/content_main" />

    <android.support.design.widget.FloatingActionButton
        android:id="@+id/fab"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_gravity="bottom|end"
        android:layout_margin="@dimen/fab_margin"
        android:src="@android:drawable/ic_dialog_email" />

</android.support.design.widget.CoordinatorLayout>
```

**Comment by sendtodilanka** [comment]
Using below code i fixed the DrawerLayout section.. but steel not fixed MiniDrawer layout. 
```
//remove the recyclerView from the slider
result.getSlider().removeView(result.getRecyclerView());
 //create new linearLayout which will contain the recyclerView
 LinearLayout ll = new LinearLayout(this);
//center the recyclerView within this container
ll.setGravity(Gravity.CENTER);
 //create new LinearLayout and set width and height
LinearLayout.LayoutParams layoutParams =  new LinearLayout
                .LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,     ViewGroup.LayoutParams.MATCH_PARENT);
//set LinearLayout margin
layoutParams.setMargins(0,60,0,0);
//add back the recyclerView in the new linearLayout container
 ll.addView(result.getRecyclerView(),layoutParams);
//add the new linearLayout container to the drawer slider
result.getSlider().addView(ll, new RelativeLayout
                .LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
```

**Comment by mikepenz** [comment]
@sendtodilanka the problem is that you add the `Crosssfader` as the parent of the `main_content` view, which is a direct child of your `CoordinatorLayout`. 

As the `CoordinatorLayout` is similar to the `FrameLayout` and does overlap views your `Crossfader` will simply be displayed below the `AppBarLayout` which you have defined to be over the `main_content`. 

To get the `Crossfader` below the `Toolbar` you should add a new `View` around your `main_content`, let's choose a `FrameLayout` which has the attribute (don't forget to define the `appBarLayout` id to your `AppBarLayout`)
```xml
android:layout_below="@id/appBarLayout"
```

After this the `Crossfader` will get inflated as a child of the `FrameLayout` we added above, which is displayed below the `AppBarLayout`.

**Comment by mikepenz** [comment]
Also answered here: http://stackoverflow.com/questions/38676461/how-to-implement-minidrawer-using-mikepenz-materialdrawer/38706600#38706600

**Comment by sendtodilanka** [comment]
@mikepenz the problem not yet solved. I edited code as your answer. but it result not change.

activity_main.xml

    <android.support.design.widget.AppBarLayout
		android:layout_width="match_parent"
		android:layout_height="wrap_content"
		android:id="@+id/appBarLayout"
		android:theme="@style/AppTheme.AppBarOverlay">

		<android.support.v7.widget.Toolbar
			android:id="@+id/toolbar"
			android:layout_width="match_parent"
			android:layout_height="?attr/actionBarSize"
			android:background="?attr/colorPrimary"
			app:popupTheme="@style/AppTheme.PopupOverlay" />

	</android.support.design.widget.AppBarLayout>

	<FrameLayout
		android:layout_width="match_parent"
		android:layout_height="match_parent"
		android:id="@+id/crossFader"
		android:layout_below="@id/appBarLayout">

		<include layout="@layout/content_main" />
	</FrameLayout>

MainActivity.java

    crossFader = new Crossfader()
		.withContent(findViewById(R.id.main_content))
		.withFirst(result.getSlider(), firstWidth)
		.withSecond(miniResult.build(this), secondWidth)
		.withSavedInstance(savedInstanceState)
		.build();

content_main.xml

    <?xml version="1.0" encoding="utf-8"?>
	<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
		xmlns:app="http://schemas.android.com/apk/res-auto"
		xmlns:tools="http://schemas.android.com/tools"
		android:layout_width="match_parent"
		android:layout_height="match_parent"
		android:id="@+id/main_content"
		android:paddingBottom="@dimen/activity_vertical_margin"
		android:paddingLeft="@dimen/activity_horizontal_margin"
		android:paddingRight="@dimen/activity_horizontal_margin"
		android:paddingTop="@dimen/activity_vertical_margin"
		app:layout_behavior="@string/appbar_scrolling_view_behavior"
		tools:context="com.sldroids.minidrawer_v3.MainActivity"
		tools:showIn="@layout/activity_main">

		<TextView
		    android:layout_width="wrap_content"
		    android:layout_height="wrap_content"
		    android:text="Hello World!" />
	</RelativeLayout>

**Comment by mikepenz** [comment]
@sendtodilanka should be fine. A similar but more simple example is the `MiniDrawerActivity` of the sample app which has the following layout: https://github.com/mikepenz/MaterialDrawer/blob/develop/app/src/main/res/layout/activity_mini_drawer.xml

**Comment by sendtodilanka** [comment]
When I used the RelativeLayout instead of CoordinatorLayout it show result as a below picture.

![screenshot_2016-08-03-15-12-45](https://cloud.githubusercontent.com/assets/20735375/17361521/87f4e154-598e-11e6-8cae-a1aa054a6773.png)


activity_main.xml

	<RelativeLayout
		xmlns:android="http://schemas.android.com/apk/res/android"
		android:layout_width="match_parent"
		android:layout_height="match_parent"
		xmlns:app="http://schemas.android.com/apk/res-auto"
		android:gravity="center">

		<android.support.design.widget.AppBarLayout
		    android:layout_width="match_parent"
		    android:layout_height="wrap_content"
		    android:id="@+id/appBarLayout"
		    android:theme="@style/AppTheme.AppBarOverlay">

		    <android.support.v7.widget.Toolbar
		        android:id="@+id/toolbar"
		        android:layout_width="match_parent"
		        android:layout_height="?attr/actionBarSize"
		        android:background="?attr/colorPrimary"
		        app:popupTheme="@style/AppTheme.PopupOverlay" />

		</android.support.design.widget.AppBarLayout>

		<FrameLayout
		    android:layout_width="match_parent"
		    android:layout_height="match_parent"
		    android:layout_below="@id/appBarLayout">

		    <include layout="@layout/content_main"/>
		</FrameLayout>
	</RelativeLayout>
as a result the miniDrawer goes to below toolbar but the color of status bar is turn to white color.. how can i fix it?

**Comment by mikepenz** [comment]
@sendtodilanka oh sorry. It does not have the `layout_below` flag but a different flag. Head over here:
http://stackoverflow.com/questions/32956071/add-views-below-toolbar-in-coordinatorlayout

**Comment by yamsergey** [comment]
Hi. First of all, great job!

But, I didn't get why the issue is closed, there are no proper solution neither here or by the links, only some discussions (I don't believe that replacing CoordinatorLayout with relative layout is an option). From the discussion I got the idea that first sibling for  `AppBarLayout` have to have option `app:layout_behavior="@string/appbar_scrolling_view_behavior"`, and since your `com.mikepenz.crossfader.view.CrossFadeSlidingPaneLayout ` wrap the content I just copy past `crossfader_base` layout and added the option to it:
```
<com.mikepenz.crossfader.view.CrossFadeSlidingPaneLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:id="@+id/pane"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    app:layout_behavior="@string/appbar_scrolling_view_behavior">
```
After I added the layout to `Crossfader` by 
`new Crossfader().withBaseLayout(R.layout.crossfader_action_bar)`
Then it starts work properly.
